tls/s2n_server_hello.c

```diff
@@ -60,10 +60,6 @@ static int s2n_hello_retry_validate(struct s2n_connection *conn) {
 }
 
 static int s2n_client_detect_downgrade_mechanism(struct s2n_connection *conn) {
-    if (!s2n_is_tls13_enabled()) {
```

Comment:
should there be a check for actual protocol?

Reply:
It's unnecessary here. The actual protocol can't be set higher than our protocol version, and we check our protocol version before actually doing anything: https://github.com/awslabs/s2n/blob/bd5aaa12874d1a18d3bfba174c708041423d950a/tls/s2n_server_hello.c#L67-L71

This behavior has also been tested by anyone who enabled TLS1.3 support, but downgraded a connection to TLS1.2 via security policies :)